airbyte-integrations/connectors/source-airtable/source_airtable/auth.py

```diff
@@ -0,0 +1,56 @@
+#
+# Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+#
+
+from typing import Any, Mapping, Union
+
+import requests
+from airbyte_cdk.sources.streams.http.requests_native_auth import (
+    BasicHttpAuthenticator,
+    SingleUseRefreshTokenOauth2Authenticator,
+    TokenAuthenticator,
+)
+
+
+class AirtableOAuth(SingleUseRefreshTokenOauth2Authenticator):
+    """
+    https://airtable.com/developers/web/api/oauth-reference#token-expiry-refresh-tokens
+    """
+
+    def build_refresh_request_headers(self) -> Mapping[str, Any]:
+        """
+        https://airtable.com/developers/web/api/oauth-reference#token-refresh-request-headers
+        """
+        config_creds = self._connector_config["credentials"]
+        return {
+            "Authorization": BasicHttpAuthenticator(config_creds["client_id"], config_creds["client_secret"]).token,
```

Comment:
Try to use ready functions: [get_client_id](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py#L184) and [get_client_secret](https://github.com/airbytehq/airbyte/blob/master/airbyte-cdk/python/airbyte_cdk/sources/streams/http/requests_native_auth/oauth.py#L187).

Reply:
Perfectly fitted)